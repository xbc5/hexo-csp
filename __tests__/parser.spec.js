const template = require("./helpers/template");
const { inject } = require("../lib/parser");

describe("for inject(html, metaTag)", () => {
  it("it should inject the meta tag into the head element", async () => {
    const before = template.basic();
    const meta = '<meta http-equiv="" content="">';

    expect(inject(before, meta)).toMatchSnapshot();
  });

  it("it should inject multiple meta tags", async () => {
    const before = template.basic();
    const meta = '<meta http-equiv="" content="">';

    expect(inject(before, meta, meta)).toMatchSnapshot();
  });
});
