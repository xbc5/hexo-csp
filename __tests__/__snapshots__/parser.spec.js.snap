// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for inject(html, metaTag) it should inject multiple meta tags 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><meta http-equiv=\\"\\" content=\\"\\"><meta http-equiv=\\"\\" content=\\"\\">
    <title>Fake Title</title>
  </head>
  <body>
</body></html>"
`;

exports[`for inject(html, metaTag) it should inject the meta tag into the head element 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><meta http-equiv=\\"\\" content=\\"\\">
    <title>Fake Title</title>
  </head>
  <body>
</body></html>"
`;
